src/main/java/com/minecolonies/coremod/entity/citizenhandlers/CitizenHappinessHandler.java

```diff
@@ -0,0 +1,596 @@
+     *
+     * @param pos
+     *            position of the scarecrow block
+     * @param canFarm
+     *            boolean to indicate if the field can be farmed
+     */
+    public void setNoFieldForFarmerModifier(final BlockPos pos, final boolean canFarm)
+    {
+        FieldDataModifier field = fieldModifier.get(pos);
+        if (field == null)
+        {
+            field = new FieldDataModifier();
+            fieldModifier.put(pos, field);
+        }
+
+        field.isCanFarm(canFarm);
+    }
+
+    /**
+     * Indicates the farmer has not fields to farm.
+     */
+    public void setNoFieldsToFarm()
+    {
+        hasNoFields = true;
+    }
+
+    /**
+     * Call this function to add a tool type that is needed by the citizen.
+     * If citizen gets its tool, then passing in false will remove it from the needed tools.
+     *
+     * @param toolType Tooltype to indicate
+     * @param needs indicate if the tool type is needed
+     */
+    public void setNeedsATool(@NotNull final IToolType toolType, final boolean needs)
+    {
+        if (needs)
+        {
+            if (!needsTool.containsKey(toolType))
+            {
+                needsTool.put(toolType, 0);
+            }
+        }
+        else
+        {
+            needsTool.remove(toolType);
+        }
+    }
+
+    /**
+     * @param hasHouse
+     *            indicate the citizen has an assigned house
+     */
+    public void setHomeModifier(final boolean hasHouse)
+    {
+        if (hasHouse)
+        {
+            houseModifier = HOUSE_MODIFIER_POSITIVE;
+        }
+        else
+        {
+            houseModifier = (MAX_HOUSE_PENALTY * ((double) numberOfDaysWithoutHouse / MAX_DAYS_WITHOUT_HOUSE)) * -1;
+            citizen.markDirty();
+        }
+    }
+
+    /**
+     * Called to set if the citizen has a job or not.
+     *
+     * @param hasJob
+     *            boolean to indicate if the citizen has a job
+     */
+    public void setJobModifier(final boolean hasJob)
+    {
+        if (hasJob)
+        {
+            jobModifier = JOB_MODIFIER_POSITIVE;
+        }
+        else
+        {
+            jobModifier = (MAX_JOB_PENALTY * ((double) numberOfDaysWithoutHouse / MAX_DAYS_WITHOUT_JOB)) * -1;
+        }
+    }
+
+    /**
+     * @return current citizens overall happiness
+     */
+    public double getHappiness()
+    {
+        double value = baseHappiness + foodModifier + damageModifier + houseModifier + jobModifier + farmerModifier + noToolModifier;
+        if (value > MAX_HAPPINESS)
+        {
+            value = MAX_HAPPINESS;
+        }
+        else if (value < MIN_HAPPINESS)
+        {
+            value = MIN_HAPPINESS;
+        }
+
+        return value;
+    }
+
+    /**
+     * @return current food modifier for happiness
+     */
+    public double getFoodModifier()
+    {
+        return foodModifier;
+    }
+
+    /**
+     * @return current damage modifier for happiness
+     */
+    public double getDamageModifier()
```

Comment:
### Errors:
- 'public' method getDamageModifier() is not exposed via an interface